flang-rt/lib/runtime/assign.cpp

```diff
@@ -279,13 +279,15 @@ RT_API_ATTRS int AssignTicket::Begin(WorkQueue &workQueue) {
     if (mustDeallocateLHS) {
       // Convert the LHS into a temporary, then make it look deallocated.
       toDeallocate_ = &tempDescriptor_.descriptor();
-      persist_ = true; // tempDescriptor_ state must outlive child tickets
```

Comment:
The general idea here is that by delaying conditionally setting persist, we can avoid persisting the allocation under some conditions and avoid the overhead of a trip through the work queue in those cases, correct?

Reply:
Actually, it was being set unconditionally, so it would be true even when the child task was run immediately as `BeginDestroy` or whatever, and that caused a later no-op suspension and resumption.